src/init.cpp

```diff
@@ -911,6 +911,15 @@ void InitParameterInteraction()
         // masternodes MUST accept connections from outside
         gArgs.ForceSetArg("-listen", "1");
         LogPrintf("%s: parameter interaction: -masternodeblsprivkey=... -> setting -listen=1\n", __func__);
+        // masternodes MUST have transaction index enabled
+        gArgs.ForceSetArg("-txindex", "1");
```

Comment:
Isn't txindex on by default? Should this log message only occur if they try to disable txindex on a MN?

Reply:
Good point. In fact our code here was already doing the same for `listen` for example which is not quite right. I refactored the whole thing, pls see 6ead5e3